### ISSUE TITLE
CSS Variables are lost when running ng build --prod

### ISSUE THREAD

**Issue by neofuture** [angular/angular#33566]
When running ng build --prod in release candidate of Angular 9 the CSS variables are removed from some elements please see screenshots  these are originally in styles.css

Development Mode
<img width="2226" alt="Screenshot 2019-11-04 at 11 08 27" src="https://user-images.githubusercontent.com/11268634/68116879-cc52d780-fef3-11e9-8263-58a3185a7f04.png">

Production Mode
<img width="2226" alt="Screenshot 2019-11-04 at 11 08 55" src="https://user-images.githubusercontent.com/11268634/68116906-d83e9980-fef3-11e9-95db-c91981cde41e.png">

**Comment by neofuture** [comment]
Work around seems to be changing 
  
  border-top:1px solid var(--background-grey) !important;

to 

  border-top:1px solid !important;
  border-color:var(--background-grey) !important;

**Comment by neofuture** [comment]
Here is a repo to illustrate the issue

https://github.com/neofuture/css-issue

**Comment by IgorMinar** [comment]
This is either a compiler or one of the optimization passes in CLI.

@neofuture does this reproduce with v8 as well?

**Comment by neofuture** [comment]
Yes the same thing happens in v8

**Comment by IgorMinar** [comment]
I don't believe that this has anything to do with the compiler because it's `--prod` mode specific. I'm going to transfer this over to the Angular CLI repo.
